crypto/asn1/evp_asn1.c

```diff
@@ -95,21 +117,64 @@ int ASN1_TYPE_get_int_octetstring(const ASN1_TYPE *a, long *num,
     if (atmp == NULL)
         goto err;
 
-    if (num != NULL)
-        *num = atmp->num;
+    ret = asn1_type_get_int_oct(atmp->oct, atmp->num, num, data, max_len);
 
-    ret = ASN1_STRING_length(atmp->oct);
-    if (max_len > ret)
-        n = ret;
-    else
-        n = max_len;
-
-    if (data != NULL)
-        memcpy(data, ASN1_STRING_get0_data(atmp->oct), n);
     if (ret == -1) {
  err:
         ASN1err(ASN1_F_ASN1_TYPE_GET_INT_OCTETSTRING, ASN1_R_DATA_IS_WRONG);
     }
     M_ASN1_free_of(atmp, asn1_int_oct);
     return ret;
 }
+
+typedef struct {
+    ASN1_OCTET_STRING *oct;
+    int32_t num;
+} asn1_oct_int;
+
```

Comment:
Please add a reference to the documentation 
as well as a listing of the ans1 data that this is need for..

Reply:
Do you mean like a comment? This is basically just like `asn1_int_oct` above but swapped. I don't see any reference for `asn1_int_oct` elsewhere in the code.

Reply:
Yes a comment: I mean an RFC document and description of section where it is referenced from.
It makes it easier for someone looking at this later to know what it is doing.

Reply:
Pushed - please check if the format is fine?

Reply:
Ah it wasn't pushed but it's pushed now. :)